datajunction-server/datajunction_server/api/nodes.py

```diff
@@ -206,6 +207,36 @@ def list_nodes(
     ]
 
 
+@router.get("/nodes/details/", response_model=List[NodeMinimumDetail])
+def list_nodes_with_details(
+    node_type: Optional[NodeType] = None,
+    *,
+    session: Session = Depends(get_session),
+    current_user: Optional[User] = Depends(get_current_user),
+    validate_access: access.ValidateAccessFn = Depends(  # pylint: disable=W0621
+        validate_access,
+    ),
+) -> List[NodeMinimumDetail]:
+    """
+    List the available nodes.
+    """
+    nodes_query = select(Node).where(
+        Node.current_version == NodeRevision.version,
+        Node.name == NodeRevision.name,
+        Node.type == node_type if node_type else True,
+        is_(Node.deactivated_at, None),
+    )
+    nodes = session.exec(nodes_query).all()
```

Comment:
All the nodes?! You think this is not going to cause us trouble? Maybe not... but then let's be explicit about it and call it `def list_all_the_nodes_with_details(`.

Reply:
You can add caching already using some standard python libs no? Say https://docs.python.org/3/library/functools.html ..